Feat/procedural skybox

nishant-kaush opened the pull request:
## What does this PR change?

<!--
In case you are fixing any specific issue, please refer to it with `Fixes #issue_number`.
In case you are implementing a new feature, please write a detailed description about it.
As an optional step, you can link or add any useful external documentation to give more context about the proposed changes (for example: design/architecture documents, figma links, screenshots, etc.).
-->

...

## How to test the changes?

<!--
Explain how to test the feature (or fix) for someone who doesn't know anything about this implementation:
At very least add the specific URL from which to test the build and add to it any param you think it would be needed.
-->

1. Go to: https://play.decentraland.zone/?renderer-branch={branch_name}&{desired_url_params}
2. ...

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

JuanMartinSalice committed:
Procedural skybox initial setup

Created the base folder for the procedural skybox files and imported the main shader files.

JuanMartinSalice committed:
Distance masking for plannar layers. Top/Bottom masking for radial layers. Changed opacity values interaction

Plannar layers now can be masked by distance with a new "RenderDistance" value. Same value can be use to mask the top/bottom of radial layers. All values that affect opacity have been reconnected in a new way to avoid undesired results.

nishant-kaush committed:
Feat: Procedural Skybox Editor tool Initial commit with as feature (Default Off)

JuanMartinSalice committed:
Stars texture. Shader adjustments

Texture created for the stars. Radial layer adjustments

nishant-kaush committed:
Merge branch 'feat/ProceduralSkybox' of https://github.com/JuanMartinSalice/unity-renderer into feat/ProceduralSkybox

JuanMartinSalice committed:
Cubemap support for the shader

Added cubemap support for the shader. When "IsRadial" value is 2 the cubemap is displayed, replacing the texture. Most of the layer properties are applied to the cubemap.

nishant-kaush committed:
feat: Integration of procedural skybox with render profile system.

JuanMartinSalice committed:
Added clouds texture

Added clouds texture. Experimentation to avoid texture cutout issues.

nishant-kaush committed:
chore: Material max number added

nishant-kaush committed:
feat: Play/Pause time from the editor to do for runtime changes and also editor time changes

nishant-kaush committed:
bug: empty scene null reference issue fixed

nishant-kaush committed:
fix: broken tool values (normal Intensity, speed)

JuanMartinSalice committed:
Moon and clouds textures. Bug fixing. Single tile development

Added moon and clouds textures. Fixed bugs related to the render distance. Single tile option testing and experimentation.

nishant-kaush committed:
feat: cubemap feature added in tool
refactor: All skybox code is under namespace DCL.Skybox
refactor: Include DCL.Skybox namespace in Plugin class

nishant-kaush committed:
feat: Render distance integration in editor tool (slider value)

nishant-kaush committed:
refactor: changed clamped values to sliders

nishant-kaush committed:
fix: Render distance issue list null references

JuanMartinSalice committed:
Clouds textures added. Cubemap Sampler Sub-Graph. Skybox Ver 03. Satellite layer

2 clouds textures were added. The cubemap layer mode now it's in a different Sub-Graph.  A basic layer for a 3rd version of the skybox it's currently in development. Satellite layer type in development.

JuanMartinSalice committed:
Satelite layer. Rebuild of Basic layer.

Satelite layer first iteration completed. Rebuild of the layers core sub-graph. "IsRadial" property replaced with "LayerType" property.

nishant-kaush committed:
chore: - Changed cubemap index as per shader change
- layer type variable name changed
- Extra "go to" button is added for float transitions. Button(>)
- Quaternion is changed to Vector3 for UI
- Horizon layer color position chnaged in tool

JuanMartinSalice committed:
Horizon mask. Rotation of textures. Different speed for each axis. New test with satellite layer

Added a texture to mask the horizon, with a values to control the size and the offset. Added a value to rotate the textures of the layers. Now the panning speed of the layers are in separated X and Y values, and are no longer multiplied with the offset. New test for the satellite layers in order to get better results when rotating.

JuanMartinSalice committed:
Merge branch 'feat/ProceduralSkybox' of https://github.com/JuanMartinSalice/unity-renderer into feat/ProceduralSkybox

nishant-kaush committed:
chore: added start and end time infront of gradient fields

JuanMartinSalice committed:
Fixed stretching of satellite layer. Cubemap rotation in other angles. Cleanup

Fixed stretching issues in the satellite layer, now each axis size can be change individually. Cubemap layers now also can be rotated in the X and Z axes. Deleted useless nodes and cleanup of the graph

nishant-kaush committed:
Merge branch 'feat/ProceduralSkybox' of https://github.com/JuanMartinSalice/unity-renderer into feat/ProceduralSkybox

JuanMartinSalice committed:
Noise deformation initial implementation. Speed in negative values

First iteration of noise deformation for layers. Added support for negative values for layer movement speed.

nishant-kaush committed:
chore: changed all transition values time span to layer time, other minor UI fixes.

JuanMartinSalice committed:
Noise deformation refinement

General improvements to noise generation. Collapsed noise deformation behavior into a Sub-Graph. Added required properties to control the noise. Added 2 types of deformation to the layer using the noise.

JuanMartinSalice committed:
Normals from textures intial setup. Particles layer intial experimentation. Deformation experimentation

Added required nodes to create a normal map from a common texture in shader. Initial experimentation of particles layers approaches. Continued experimentation and refinement of deformation for layers

JuanMartinSalice committed:
Distortion cleanup. Layer particles experimentation.

Cleanup of distortion for layers. Unused nodes removed. Added textures which uses RGB for normals and A for distortion. Experimentation and testing of different approaches for the particle layers.

nishant-kaush committed:
refactor: code refactoring for slot system
chore: Added satellite layer and horizon mask

JuanMartinSalice committed:
Added atlas textures. Added missing distortion parameters. Particles layer sub-graph

Added an atlas texture to be tested as a texture array 2D for saving texture space. Added distortion parameters to the layers 1 and 2. Created a particle layer sub-graph with advance development in the feature.

nishant-kaush committed:
- slot system completed
- removed rotation from radial layer

JuanMartinSalice committed:
Merge branch 'feat/ProceduralSkybox' of https://github.com/JuanMartinSalice/unity-renderer into feat/ProceduralSkybox

nishant-kaush committed:
bug of horizon mask texture

JuanMartinSalice committed:
Bug fixes

Cubemap rotation not in degrees fix

JuanMartinSalice committed:
Merge branch 'feat/ProceduralSkybox' of https://github.com/JuanMartinSalice/unity-renderer into feat/ProceduralSkybox

nishant-kaush committed:
change:Percentage part of trasitioning values is changed to a more info one.

nishant-kaush committed:
Merge branch 'feat/ProceduralSkybox' of https://github.com/JuanMartinSalice/unity-renderer into feat/ProceduralSkybox

JuanMartinSalice committed:
Particles layer integration and development

Further development of the particle layer feature. Integrated particle layer to the rest of the layer system. Added timed fade of particles

JuanMartinSalice committed:
Merge branch 'feat/ProceduralSkybox' of https://github.com/JuanMartinSalice/unity-renderer into feat/ProceduralSkybox

JuanMartinSalice committed:
Added slots. Refactor of layers. Advances in particle layer

Added two more slots to the main shader, new maximum of five. Refactoring of layers to avoid unnecessary texture samples (removed texture sample from distortion). Improvements and cleanup of particles sub-graph.

JuanMartinSalice committed:
Added parameters for particle layers and bug fixes

Added parameters to control the particle layers

nishant-kaush committed:
feat:
- enable/disable of layers and slots
- UX improvements on moving layers between slots

JuanMartinSalice committed:
Particles parameters organized. Fading of particles. Bug fixing

Reorganized parameters for the particles. Particles fading and changing position. Fixed visibility bugs related to the particles working in other layer types.

JuanMartinSalice committed:
Merge branch 'feat/ProceduralSkybox' of https://github.com/JuanMartinSalice/unity-renderer into feat/ProceduralSkybox

nishant-kaush committed:
- Particle Layer variables added
- Other monior UI fixed

nishant-kaush committed:
- UX Improvements
- Particle layer option in each kind of layer except cubemap layer

nishant-kaush committed:
- Integration with main environment and variables exposed to change configuration file and speed of time

JuanMartinSalice committed:
Particle Layer

Final particle layer added with all of the requested functionalities

nishant-kaush committed:
New Variables related to particles

nishant-kaush committed:
Time spanning over a day for any layer now working. Ex 22:00 to 07:00

JuanMartinSalice committed:
Removed time control from the shader

Removed time control from the layers, now it should be only controlled by the tool

nishant-kaush committed:
Context menu button not required for changing at runtime. It will reflect at realtime
Only one option will be visible now either speed or offset/position for movement of the layer.

JuanMartinSalice committed:
Bug fixes and test scene created

Fixed satellite masking bugs and general optimization of the shader. Test scene created

JuanMartinSalice committed:
Particle layer improvement and bug fixes

Greatly improved particle layer. Fixed bugs. Added Cubemap Sample as a Sub-Graph to sample 2D textures as cubemaps

JuanMartinSalice committed:
Fade time parameter in shader. Particles as new layer mode. Particles layer rotation. Cleanup

Fade time parameter in shader now controls the opacity of the whole layer. Particles layer is now a new layer mode, instead of a layer feature. Particles layer can now rotate as the cubemap layer. Cleanup of unused nodes and better layer shader flow.

nishant-kaush committed:
- Particle layer added.
- Particle is removed from other layers

nishant-kaush committed:
Particle layer bug fixed

JuanMartinSalice committed:
Minor changes

Minor changes and optimization of the shader.

nishant-kaush committed:
- Over the day transition bug fixed
- Minor UI Polishing
- Code refactoring

nishant-kaush committed:
Merge branch 'feat/ProceduralSkybox' of https://github.com/JuanMartinSalice/unity-renderer into feat/ProceduralSkybox

nishant-kaush committed:
- fading is removed from tool for now and hard coded to 1
- Layer type resets issue fixed

JuanMartinSalice committed:
Added flipbook almost all layer types. Fixed day clouds cubemap

Added a flipbook feature to planar, radial and satellite layer types. Fixed day clouds cubemap visual errors.

nishant-kaush committed:
- Rotation value for particle layer is changed to transitioning value
- Flipbook animation is added to all layers except cubemap

nishant-kaush committed:
Fading feature is added for normal and over the day scenarios

JuanMartinSalice committed:
Shader cleanup

Removed Time related unused nodes from the sahder.

JuanMartinSalice committed:
Fixed normals and better tint

Fixed layer normals bug. Improve light blending on normals

JuanMartinSalice committed:
Added 3D assets for Skybox 3D layers

Added a cloud mesh. Added a star mesh. Created a particle system to spawn the cloud mesh. Created a trail and particles system to create a shooting star with the star mesh. Created a cloud shader to create procedural volumetric clouds on a plane.

nishant-kaush committed:
Merge branch 'master' into feat/Procedural_skybox

# Conflicts:
#	unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FeatureController/FeatureController.asmdef
#	unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FeatureController/PluginSystem.cs
#	unity-renderer/Assets/Scripts/MainScripts/DCL/DataStore/DataStore.cs
#	unity-renderer/Assets/Scripts/MainScripts/DCL/KernelConfiguration/KernelConfigTypes.cs

nishant-kaush committed:
New feature flag implementation for procedural skybox.

nishant-kaush committed:
Skybox Editor code is moved to editor assembly

BrianAmadori commented on unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/SkyboxController.cs:
```diff
@@ -0,0 +1,200 @@
+            DataStore.i.skyboxConfig.objectUpdated.Set(false, false);
+        }
+
+        void Applyconfig()
```
```suggestion
        void ApplyConfig()
```

nishant-kaush replied:
Done. Making all these changes in a new commit as all the calls need to be changed too.

BrianAmadori commented on unity-renderer/Assets/Scripts/MainScripts/DCL/DataStore/DataStore.cs:
```diff
@@ -152,7 +153,18 @@ public class DataStore_FeatureFlag
+            public BaseVariable<string> configToLoad = new BaseVariable<string>("Generic Skybox");
+            public BaseVariable<float> minutesPerSecond = new BaseVariable<float>(60);
+            public BaseVariable<bool> pauseTime = new BaseVariable<bool>(false);
+            public bool jumpTime = false;
+            public BaseVariable<float> jumpToTime = new BaseVariable<float>(0);
```
Why do we have `jumpTime` and `jumpToTime`? Please change the name or remove the variable if its not used.

nishant-kaush replied:
- jumpToTime (BaseVariable<float>) is used when we have to jump to a specific time of the day... can provide float value of time.
- jumpTime (bool) is a flag after which "jumpToTime" is processed. 

Bool is required to assure if the value in "jumpToTime" should be applied or not, else it can lead to some bugs.

BrianAmadori commented on unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/SkyboxController.cs:
```diff
@@ -0,0 +1,200 @@
+        public override void Dispose()
+        {
+            base.Dispose();
+            //DCL.DataStore.i.isProceduralSkyboxInUse.Set(false);
```
```suggestion
```

nishant-kaush replied:
Done. Making all these changes in a new commit

BrianAmadori commented on unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/SkyboxController.cs:
```diff
@@ -0,0 +1,200 @@
+            {
+                directionalLight.gameObject.SetActive(false);
+            }
+            //DCL.DataStore.i.isProceduralSkyboxInUse.Set(true);
```
```suggestion
```

nishant-kaush replied:
Done. Making all these changes in a new commit

BrianAmadori commented on unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/Editor/Procedural Skybox/SkyboxEditor.cs:
```diff
@@ -0,0 +1,1344 @@
+        {
+
+            SkyboxEditorWindow window = (SkyboxEditorWindow)EditorWindow.GetWindow(typeof(SkyboxEditorWindow));
+            //SkyboxEditorWindow.selectedConfiguration = new SkyboxConfiguration();
```
```suggestion
```

nishant-kaush replied:
Done. Making all these changes in a new commit

BrianAmadori commented on unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/Editor/Procedural Skybox/SkyboxEditor.cs:
```diff
@@ -0,0 +1,1344 @@
+            selectedMat = matLayer.material;
+            selectedConfiguration.ResetMaterial(selectedMat, matLayer.numberOfSlots);
+            RenderSettings.skybox = selectedMat;
+            //initialized = true;
```
```suggestion
```

nishant-kaush replied:
Done. Making all these changes in a new commit

BrianAmadori commented on unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/Editor/Procedural Skybox/SkyboxEditor.cs:
```diff
@@ -0,0 +1,1344 @@
+
+            InitializeMaterial();
+
+            //isPaused = true;
```
```suggestion
```

Remember that all commented code should be removed. For reverting code we already have the commits history!.

nishant-kaush replied:
Done. Making all these changes in a new commit

BrianAmadori commented on unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/Editor/Procedural Skybox/SkyboxEditor.cs:
```diff
@@ -0,0 +1,1344 @@
+        void RenderBackgroundColorLayer()
+        {
+
+            //GUILayout.Label("Background Layer", EditorStyles.boldLabel);
```
```suggestion
```

nishant-kaush replied:
Done. Making all these changes in a new commit

BrianAmadori commented on unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/Editor/Procedural Skybox/SkyboxEditor.cs:
```diff
@@ -0,0 +1,1344 @@
+        void RenderAmbientLayer()
+        {
+
+            //GUILayout.Label("Ambient Layer", EditorStyles.boldLabel);
```
```suggestion
```

nishant-kaush replied:
Done. Making all these changes in a new commit

BrianAmadori commented on unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/Editor/Procedural Skybox/SkyboxEditor.cs:
```diff
@@ -0,0 +1,1344 @@
+            RenderSlots();
+
+            // Render Texture Layers
+            //RenderTextureLayers();
```
If `RenderTextureLayers` is not needed, we should remove the method altogether.

nishant-kaush replied:
Done. Making all these changes in a new commit

BrianAmadori reviewed (changes requested):
Leaving here the first code review. Good work! 🍻

JuanMartinSalice committed:
Added 3D clouds variations. Trail shader. 3D Cloud shader. Bug fixes

Added more 3D clouds meshes and integrated them to particle system. Created shader to distort the 3D clouds. Created a trail shader for shooting star. Added secondary camera to screen. Fixed horizon mask bug.

nishant-kaush committed:
Merge branch 'master' into feat/Procedural_skybox

nishant-kaush committed:
feat: avatar colors integration in the tool
- code review related changes, mostly commented code and extra variables removed

nishant-kaush committed:
fix: skybox shader compile issue fixed.

nishant-kaush committed:
fix: fog is enabled by default with high fog values for character colors

JuanMartinSalice committed:
Minor changes to 3D assets

Reduced amount of poligons on 3D clouds. Drasticaly reduced the amount of poligons of 3D plane used for clouds. Added 3D dome. Improved cloud shader.

nishant-kaush committed:
Merge branch 'master' into feat/Procedural_skybox

JuanMartinSalice committed:
Dome Shaders

Created 3 shaders for the dome mesh. One sound bands shader. One simple cubemap shader. One texture tiling and moving shader.

nishant-kaush committed:
Configuration (Variation_1) from Alvaro

JuanMartinSalice committed:
Particles shader for the dome. Satellite shader for the dome. Added movement to cubemap

Created a particle shader for the dome mesh. Created a satellite shader for the dome mesh. Added movement to the cubemap shader for the cube.

JuanMartinSalice committed:
Merge branch 'feat/Procedural_skybox' of https://github.com/JuanMartinSalice/unity-renderer into feat/Procedural_skybox

nishant-kaush committed:
chore: configuration support is added. Now Configuration value such as configToLoad and minutesPerSecond can be changed through kernel config

JuanMartinSalice committed:
Improved dome shaders. Added Shooting stars particles.

Improved all dome shaders. Added a new particle effect for shooting stars.

nishant-kaush committed:
Object field is added for selection of skybox config in the tool

nishant-kaush committed:
changes:
- minutes per seconds is changed to lifecycle Duration in minutes
- layers checkbox working improved
- Skybox is enabled by default for this branch in PluginSystem
- Default config is changed to "Variation_1" and default cycle time to 1 min

JuanMartinSalice committed:
Reorganized test scene. Planar shader for dome.

Reorganized test scene, to enable easy access to the different 3D assets. Created a planar shader for the dome, that can be used in other 3D objects

JuanMartinSalice committed:
Skybox position changes. Planar 3D object shader

Skybox features are now independent of the position. Planar shader for 3D objects such as the dome.

nishant-kaush committed:
- Time sync with the world for skybox controller
- Event system for timestamp and triggers

nishant-kaush committed:
- Timer converted to Singelton
- UI changed: Slot UI removed

JuanMartinSalice committed:
Mask below horizon. Plane color below the horizon. Reflection probe added.

Added a mask for the layers that mask everything below the horizon. Also added a plane projected on that mask. Added a color for the plane and it's opacity and a float value to control the height of the mask and the plane color. Added reflection probe to test reflections baking.

nishant-kaush committed:
Merge branch 'master' into feat/Procedural_skybox

# Conflicts:
#	unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FeatureController/FeatureController.asmdef
#	unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FeatureController/PluginSystem.cs
#	unity-renderer/Assets/Scripts/MainScripts/DCL/DataStore/DataStore.cs
#	unity-renderer/UserSettings/EditorUserSettings.asset

nishant-kaush committed:
chore: solved merge conflicts and correction regarding recent changes in project.

* Changes after merge conflict.

* Few changes in the tool as the initial doesn't have sun by default

JuanMartinSalice committed:
Reflections sample script

Added a script that sample reflections with a probe using a determined frequency.

nishant-kaush committed:
chore: added functionality in the tool

* while adding new tag take current time and make it trigger by default

* change cycle time and configuration from editor tool at runtime

* Removed word "Layer"

JuanMartinSalice committed:
Removed full transparency from skybox shader. Horizon plane value changes.

Removed transparency from skybox background which might cause visual bugs in the builds. Changed the horizon plane value to a slider and inverted the value.

JuanMartinSalice committed:
Dome skybox shader

Created a dome skybox shader, with only one slot and full transparency enabled.

nishant-kaush committed:
chore: added functionality in the tool

* GUI Styles moved

* Timeline tags moved with time panel

* Up/Down arrow added

* Marker added for rendering

* Test values removed from DebugConfig class. All testing for runtime can be handled via tool now

nishant-kaush committed:
Merge branch 'feat/Procedural_skybox' of https://github.com/JuanMartinSalice/unity-renderer into feat/Procedural_skybox

nishant-kaush committed:
chore: changed default values

* Default config is changed to Generic_Skybox

* Varition_1 config name is changed to Generic_Skybox

* Default time is now 120 mins

* Editor tool: Rendering order is changed to dropdown

nishant-kaush committed:
Merge branch 'master' into feat/Procedural_skybox

# Conflicts:
#	unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FeatureController/FeatureController.asmdef
#	unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FeatureController/PluginSystem.cs

nishant-kaush committed:
chore: skybox controller updated with new plugin system

* SkyboxController now inherits from IPlugin

* SkyboxController is bind with flag "procedural_skybox"

nishant-kaush committed:
chore: fixed time functionality and procedural_skybox is on by default

* fixed time functionality

* procedural_skybox is on by default

nishant-kaush committed:
refactor: default skybox name changed

nishant-kaush committed:
chore: flag name changed. Currently the feature is open by default

nishant-kaush committed:
fix: skybox time sync issue fixes
chore: ENABLE_PROCEDURAL_SKYBOX is required to enable the procedural skybox

nishant-kaush committed:
fix:procedural_skybox is required to enable the procedural skybox

nishant-kaush committed:
fix: skybox timer issue

* http changed to https

* system time is get as universal time

nishant-kaush committed:
feat: runtime environment reflections added

* Resources folder is moved out of script folder

* new probe prefab is added in resources

* Extra configs deleted

* new variable related to reflections is added in kernel config

nishant-kaush committed:
chore: kernel variable name updated  + debugs

* kernel config variable name changed
* more debugs to get timer issue and reflection probe issue
* editor tool path changed

nishant-kaush committed:
fix: reflection probe issue + timer issue

JuanMartinSalice committed:
New skybox configuration

Created a new skybox configuration for further testing. Deleted unused material.

JuanMartinSalice committed:
Merge branch 'feat/Procedural_skybox' of https://github.com/JuanMartinSalice/unity-renderer into feat/Procedural_skybox

nishant-kaush committed:
chore: debug value added for procedural skybox

nishant-kaush committed:
chore: debug value added and timer code changed + timer paused (testing)
*  debug value added for procedural skybox : UTC Time calculated
* timer logic changed

olavra committed:
Changed generic skybox configuration

nishant-kaush committed:
fix: timer clamped values to 2 decimals + editor time slider width flexible + Render distance in radial layer not working issue fix

nishant-kaush committed:
Merge branch 'feat/Procedural_skybox' of https://github.com/JuanMartinSalice/unity-renderer into feat/Procedural_skybox

nishant-kaush committed:
fix: reflection probe changed to type custom

nishant-kaush committed:
fix: skybox day and night cycle resumes.

nishant-kaush committed:
chore: small hiccups removed from skybox transitions

nishant-kaush committed:
fix: timer values shifted to double for precision

nishant-kaush committed:
fix: trying custom cubemap working in webGL

nishant-kaush committed:
fix: timer api changed + reflection issue fix

* timer api changed to catalyst for now
* new cube blend shader added for reflection

nishant-kaush committed:
fix: two more shader committed related to cubemap

nishant-kaush committed:
fix: Reflection + timer
Reflection: Probe is removed from camera child and only realtime reflection is used... no custom
Timer: time will be synced with server time after every 10 frame

nishant-kaush committed:
chore: extra debugs and commented code removed.

nishant-kaush committed:
feat: Options in Settings menu + Horizon plane values in editor tool

nishant-kaush committed:
Merge branch 'master' into feat/Procedural_skybox

# Conflicts:
#	unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/PluginSystem/PluginSystem.asmdef
#	unity-renderer/Assets/Scripts/MainScripts/DCL/DataStore/DataStore.cs
#	unity-renderer/Assets/Scripts/MainScripts/Debugging/DebugParameters/DebugConfigComponent.cs
#	unity-renderer/Assets/Scripts/MainScripts/Debugging/FPSDisplay/FPSDisplay.asmdef

nishant-kaush committed:
merge: conflicted file and error commit

nishant-kaush committed:
feat: Widget in Setting HUD update. Disabled slider when skybox is dynamic

nishant-kaush committed:
feat: New controls in settings menu
* Dropdown changed to toggle
* Reflection resolution in graphics settings
* slider text is changed to HH:MM format

nishant-kaush committed:
fix:  Editor tool bugs
* when a layer is supposed to last all day, having the time of the day at hour 24 makes the layer fade away.
* And time values for the layer can be outside the 0-24 range.

nishant-kaush committed:
fix: reflection issues

* Bug fixed: reflection not changing after resolution change till relaunch
* Optimize: for fixed skybox reflection will after a cooldown of 3 sec

nishant-kaush committed:
feat: settings option for procedural skybox will deactivate with procedural_skybox feature flag

nishant-kaush committed:
fix: runtime reflection resolution changes + reflection resolution is added in PerformanceMeterController

nishant-kaush committed:
chore: avatar light color is changed to gradient from simple color

nishant-kaush committed:
Merge branch 'master' into feat/Procedural_skybox

# Conflicts:
#	unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/SettingsPanelHUD/Configuration/Widgets/GeneralWidget.asset
#	unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/TaskbarHUD/TaskbarMoreMenu.cs
#	unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/Settings/Settings.cs
#	unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/Settings/SettingsData/GeneralSettings.cs

nishant-kaush committed:
refactor: new UI related changes

* Main branch merge conflict solved
* night mode toggle removed for procedural skybox
* integration with new UI

nishant-kaush committed:
refactor: reflection resolution association with presets
* number of reflection resolution decreased

nishant-kaush committed:
refactor: code cleanup
* All commented code should be removed.
* removed underscoring for private fields or arguments, we should remove all occurrences to keep consistence across the project.
* use the DataStore to store the time value to prevent coupling.

olavra committed:
Updated generic configuration

nishant-kaush committed:
Merge branch 'master' into feat/Procedural_skybox

# Conflicts:
#	unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/Settings/Settings.cs

nishant-kaush committed:
refactor: merge changes.... default values in genral settings

olavra committed:
chore: update default skybox configuration
